src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -483,9 +660,137 @@ private async Task<ReviewListItemModel> ToggleReviewApproval(ClaimsPrincipal use
             review.ChangeHistory = changeUpdate.ChangeHistory;
             review.IsApproved = changeUpdate.ChangeStatus;
             await _reviewsRepository.UpsertReviewAsync(review);
+            
+            // If this review was approved and has pending namespace status with a timestamp, check if we should approve the TypeSpec namespace
+            if (changeUpdate.ChangeStatus && 
+                LanguageHelper.IsSDKLanguage(review.Language) && 
+                review.NamespaceReviewStatus == NamespaceReviewStatus.Pending &&
+                review.NamespaceApprovalRequestedOn.HasValue)
+            {
+                await CheckAndApproveNamespaceForTypeSpec(review);
+            }
+            
             return review;
         }
 
+        /// <summary>
+        /// Get all reviews with a specific namespace timestamp regardless of status
+        /// </summary>
+        /// <param name="namespaceTimestamp">The namespace approval requested timestamp</param>
+        /// <returns>List of reviews with the specified timestamp</returns>
+        private async Task<List<ReviewListItemModel>> GetReviewsWithNamespaceTimestamp(DateTime namespaceTimestamp)
+        {
+            try
+            {
+                // Get all reviews that have namespace approval requests
+                var allLanguages = ApiViewConstants.AllSupportedLanguages;
+                var allNamespaceReviews = await _reviewsRepository.GetPendingNamespaceApprovalReviewsAsync(allLanguages);
```

Comment:
I don't see a place in where you change the status of pending for other reviews other than TypeSpec, so I think this can become problematic as time pass by 

also, why we need to do this with timestamp?